gammapy/modeling/models/time.py

```diff
@@ -123,6 +125,74 @@ def evaluate_norm_at_phase(self, phase):
         fp = self.table["NORM"]
         return np.interp(x=phase, xp=xp, fp=fp, period=1)
 
+    #    @property
+    def ontime(self, t_min, t_max):
+        """On time (`~astropy.units.Quantity`)"""
+
+        t_min = Time(t_min)
+        t_max = Time(t_max)
+
+        ontime = t_max - t_min
+        return u.Quantity(ontime.sec, "s")
+
+    def _get_time_meta(self, t_min, t_max):
+        """Time meta information (`OrderedDict`)"""
+        # TODO: extend the meta information according to
+        #  https://gamma-astro-data-formats.readthedocs.io/en/latest/general/time.html#time-formats
+        meta = OrderedDict()
+        meta["ONTIME"] = np.round(
+            self.ontime(t_min=t_min, t_max=t_max).to_value("s"), 1
+        )
+        return meta
+
+    def sample_time(self, n_events, t_min, t_max, t_delta="1 s", random_state=0):
+        """Sample arrival times of events.
+
+        Parameters
+        ----------
+        n_events : int
+            Number of events to sample.
+        t_min : `~astropy.time.Time`
+            Start time of the sampling.
+        t_max : `~astropy.time.Time`
+            Stop time of the sampling.
+        t_delta : `~astropy.units.Quantity`
+            Time step used for sampling of the temporal model.
+        random_state : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}
+            Defines random number generator initialisation.
+            Passed to `~gammapy.utils.random.get_random_state`.
+
+        Returns
+        -------
+        time : `~astropy.units.Quantity`
+            Array with times of the sampled events.
+        """
+        time_unit = u.second
+
+        t_min = Time(t_min)
+        t_max = Time(t_max)
+        n_events = n_events
+        t_delta = u.Quantity(t_delta)
+        random_state = get_random_state(random_state)
+
+        t_stop = self.ontime(t_min=t_min, t_max=t_max).to_value(time_unit)
+
+        # TODO: the separate time unit handling is unfortunate, but the quantity support for np.arange and np.interp
+        #  is still incomplete, refactor once we change to recent numpy and astropy versions
+        if self.table is not None:
```

Comment:
Remove this logic of `if self.table is not None`. The case of a uniform time distribution we will handle outside. I think it's not useful to have it attached to the model.

Reply:
Please still address this comment...

Reply:
If I remove it, then the user is forced to always pass a lightcurve. Is what we really want?